bambi/models.py

```diff
@@ -746,9 +746,9 @@ def __init__(self, name, data, categorical=False, random=False, prior=None,
 
         # For the sake of computational efficiency (i.e., to avoid lots of
         # large matrix multiplications in the backends), invert the dummy-
-        # coding process and represent binary categoricals as a vector of
+        # coding process and represent full-rank dummies as a vector of
         # indices into the coefficients.
-        if data.shape[1] > 1 and ((data == 0) | (data == 1)).all():
+        if constant and data.shape[1] > 1 and ((data == 0) | (data == 1)).all():
```

Comment:
I don't think that particular tweak would work (although I admittedly haven't tried it), because the main issue is that, for the reference level, there is no valid index for indicating the non-zero column because all columns are zero for the reference level. So we would probably have to do something involving a special value (like None or a negative index) for that case to differentiate it from the case of a valid index of 0. My guess is that we could work out a solution for this if we wanted to, and did I almost try. I decided not to simply because this would only affect fixed categoricals, which will rarely have more than handful of levels anyway, so it's not clear that the compression would have any noticeable impact anyway (unlike for random effects which may have many levels).